app/controllers/application_controller.rb

```diff
@@ -179,10 +179,13 @@ def engine_content?
   end
   helper_method :engine_content?
 
-  PENSIONS_CATEGORY = 'pensions-and-retirement'
+  PENSIONS_CATEGORY = 'pensions-and-retirement'.freeze
   def pensions_and_retirement_page?
-    @article.present? &&
-      @article.categories.map(&:parent_id).include?(PENSIONS_CATEGORY)
+    check = [@category].push(*@article.try(:categories))
+    check.compact.any? do |category|
+      category.parent_id == PENSIONS_CATEGORY ||
+        category.id == PENSIONS_CATEGORY
+    end
```

Comment:
This method tends to become bigger with time with different contexts (article page, tools page and in the future category, video pages) for the same method. Since this is the second time, shall we move this to your own ruby class? Wdyt?